syntax = "proto3";

service stockSystem{
 rpc Lookup (stockName) returns (PriceVolume) {}
 rpc Trade (tradingRequest) returns (statusIndicator){}
 rpc Update (updateRequest) returns (statusIndicator){}
}

message stockName{
 string stock_name = 1;
}

message PriceVolume{
 int32 stock_price = 1;
 int32 stock_volume = 2;
}

message tradingRequest{
  string stock_name = 1;
  int32 num = 2;
  string trade_type = 3;
}

message statusIndicator{
  int32 status_indicator = 1;
}

message updateRequest{
  string stock_name = 1;
  int32 stock_price = 2;
}

